xla/service/cpu/onednn_memory_util.cc

```diff
@@ -73,17 +73,14 @@ MemrefInfoHandler CreateMemrefInfoFromLiteral(const Literal* literal) {
   return CreateMemrefFromShape(shape, buf);
 }
 
-StackAlloca GetAllocaAndEmitMemrefInfo(llvm::IRBuilderBase& builder,
-                                       const llvm_ir::IrArray& ir_array) {
-  const Shape& shape = ir_array.GetShape();
-  // oneDNN handles scalar as a vector of size 1.
-  int64_t rank = shape.rank() == 0 ? 1 : shape.rank();
+void SetDimsStrides(const Shape& shape, std::vector<int64_t>& dims,
+                    std::vector<int64_t>& strides) {
```

Comment:
Nit: We prefer returning the output over having it as a parameter. Please change?

Reply:
Done. I also renamed the function name to GetDimsStrides. Please let me know if that looks ok. Thanks!